lib/rack/lint.rb

```diff
@@ -513,129 +510,90 @@ def close(*args)
         end
       end
 
-      class HijackWrapper
-        extend Forwardable
-
-        REQUIRED_METHODS = [
-          :read, :write, :read_nonblock, :write_nonblock, :flush, :close,
-          :close_read, :close_write, :closed?
-        ]
-
-        def_delegators :@io, *REQUIRED_METHODS
-
-        def initialize(io)
-          @io = io
-          REQUIRED_METHODS.each do |meth|
-            raise LintError, "rack.hijack_io must respond to #{meth}" unless io.respond_to? meth
-          end
-        end
-      end
-
       ##
       ## === Hijacking
       ##
-      #
-      # AUTHORS: n.b. The trailing whitespace between paragraphs is important and
-      # should not be removed. The whitespace creates paragraphs in the RDoc
-      # output.
-      #
-      ## ==== Request (before status)
+      ## The hijacking interfaces provides a means for an application to take
+      ## control of the HTTP connection. There are two distinct hijack
+      ## interfaces: full hijacking where the application takes over the raw
+      ## connection, and partial hijacking where the application takes over
+      ## just the response body stream. In both cases, the application is
+      ## responsible for closing the hijacked stream.
+      ##
+      ## Full hijacking only works with HTTP/1. Partial hijacking has been
+      ## superseeded by streaming responses.
```

Comment:
It would be `superseded`, but that can also have the interpretation that partial hijacking is no longer available. I would reword to `Partial hijacking is functionally equivalent to streaming bodies, and is still optionally supported for backwards compatibility with older Rack versions.`

Reply:
Thanks, this is much better.